File: src/python/oop/FileSearch/search/file_info.py
import os

class FileInfo:
    def __init__(self, file: os.DirEntry):
        self.__file = file

    @property
    def name(self) -> str:
        if self.__file.name.startswith('.'):
            return self.__file.name

        return os.path.basename(self.__file.name).split('.')[0]

    @property
    def extension(self) -> str:
        return os.path.basename(self.__file.name).split('.')[-1]

    @property
    def size(self) -> int:
        return os.stat(self.__file.path).st_size

    @property
    def path(self) -> str:
        return self.__file.path

File: src/go/oop/FileSearch/internal/models/file_info.go
package models

import (
	"os"
	"path/filepath"
	"strings"
)

type FileInfo struct {
	Path string
	info os.FileInfo
}

func File(path string) (*FileInfo, error) {
	file, err := os.Stat(path)
	if err != nil {
		return nil, err
	}
	return &FileInfo{
		Path: path,
		info: file,
	}, nil
}

func (f *FileInfo) Name() string {
	name := f.info.Name()
	if strings.Index(name, ".") <= 0 {
		return name
	}

	return strings.TrimSuffix(name, filepath.Ext(f.Path))
}

func (f *FileInfo) Extension() string {
	return filepath.Ext(f.Path)
}

func (f *FileInfo) Size() int64 {
	return f.info.Size()
}
